gensim/downloader.py

```diff
@@ -0,0 +1,274 @@
+    url_load_file = "https://github.com/chaitaliSaini/gensim-data/releases/download/{f}/__init__.py".format(f=name)
+    data_folder_dir = os.path.join(base_dir, name)
+    compressed_folder_name = "{f}.tar.gz".format(f=name)
+    tmp_dir = tempfile.mkdtemp()
+    tmp_load_file = os.path.join(tmp_dir, "__init__.py")
+    urllib.urlretrieve(url_load_file, tmp_load_file)
+    logger.info("Downloading %s", name)
+    tmp_data_file = os.path.join(tmp_dir, compressed_folder_name)
+    urllib.urlretrieve(url_data, tmp_data_file, reporthook=progress)
+    if _calculate_md5_checksum(tmp_data_file) == _get_checksum(name):
+        logger.info("%s downloaded", name)
+    else:
+        rmtree(tmp_dir)
+        raise Exception("There was a problem in downloading the data. We recommend you to re-try.")
+    with tarfile.open(tmp_data_file, 'r') as tar:
+        tar.extractall(tmp_dir)
+    os.remove(tmp_data_file)
+    os.rename(tmp_dir, data_folder_dir)
+
+
+def _get_filename(name):
+    r"""Retrive the filename(or foldername, in case of some datasets) of the dataset/model.
+
+    Parameters
+    ----------
+    name: str
+        dataset/model of which filename is retrieved.
+
+    Returns
+    -------
+    str:
+        filename(or foldername, in case of some datasets) of the dataset/model.
+
+    """
+    data = info()
+    corpora = data['corpora']
+    models = data['models']
+    if name in corpora:
+        return data['corpora'][name]["file_name"]
+    elif name in models:
+        return data['models'][name]["file_name"]
+
+
+def load(name, return_path=False):
+    r"""For models, if return_path is False, then load model to memory. Otherwise, return the path to the model.
+    For datasets, return path to the dataset in both cases.
+
+    Parameters
+    ----------
+    name: str
+        name of the model/dataset
+    return_path:{False, True}, optional
+
+    Returns
+    -------
+    data:
+        load model to memory
+    data_dir: str
+        return path of dataset/model.
+    """
+    _create_base_dir()
+    file_name = _get_filename(name)
+    if file_name is None:
+        raise Exception(
+            "Incorrect model/corpus name. Choose the model/corpus from the list "
+            "\n {}".format(json.dumps(info(), indent=4)))
+    folder_dir = os.path.join(base_dir, name)
+    data_dir = os.path.join(folder_dir, file_name)
+    if not os.path.exists(folder_dir):
+        _download(name)
+
+    if return_path:
+        return data_dir
+    else:
+        sys.path.insert(0, base_dir)
+        module = __import__(name)
+        data = module.load_data()
+        return data
+
+
+if __name__ == '__main__':
+    logging.basicConfig(format='%(asctime)s :%(name)s :%(levelname)s :%(message)s', stream=sys.stdout, level=logging.INFO)
+    parser = argparse.ArgumentParser(description="Gensim console API", usage="python -m gensim.api.downloader  [-h] [-d data__name | -i data__name | -c]")
```

Comment:
No need to pass custom "usage" string here (argparse will generate it automatically)